### PR TITLE
FIx string_view hash function

### PR DESCRIPTION
**Pull request by dragon512**
Add a unit test

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
Did you verify the test fails without the fix?

**Comment by dragon512** [comment]
@SolidWallOfCode Yes... the original code is working based on the pointer value. I missed that the std::hash<char *> is just a passing around a pointer.. This was my mistake that was documented in a different section of the C++ spec.

File: lib/ts/string_view.h [hunk 1]
```diff
@@ -1220,7 +1220,13 @@ template <class _Type, class _Traits> struct hash<ts::basic_string_view<_Type, _
   size_t
   operator()(string_type const &x) const
   {
-    return hash<typename string_type::const_pointer>()(x.data());
+// not what I would normally do.. but better than
+// making a custom has at the moment
```
Comment (bryancall) [inline lib/ts/string_view.h]: s/has/hash ?

Comment (dragon512) [reply lib/ts/string_view.h]: @bryancall What is the question/concern?

File: lib/ts/string_view.h [hunk 2]
```diff
@@ -1220,7 +1220,13 @@ template <class _Type, class _Traits> struct hash<ts::basic_string_view<_Type, _
   size_t
   operator()(string_type const &x) const
   {
-    return hash<typename string_type::const_pointer>()(x.data());
+// not what I would normally do.. but better than making a custom hash function at the moment.
+// This should also mean we have some consistent behavior with std code
+#if defined(__linux__)
```
Comment (SolidWallOfCode) [inline lib/ts/string_view.h]: I think this is not OS dependent but compiler dependent. E.g. `__gnuc` and `__clang`.

Comment (SolidWallOfCode) [reply lib/ts/string_view.h]: Oddly, I tried this PR using clang on fedora 27 and it builds fine. Not a real concern, then.

**Comment by zwoop** [comment]
This fails on CentOS6 for some reason, with devtoolset-3 (which is gcc-4.9, our currently oldest supported compiler).

**Comment by bryancall** [comment]
Can't we use another hash function instead of using non-standard hash functions that are platform dependent?

**Comment by PSUdaemon** [comment]
We do have some good hash implementations already in the code.

> lib/ts/Hash.h
> lib/ts/HashFNV.h
> lib/ts/HashMD5.h
> lib/ts/HashSip.h

**Comment by bryancall** [comment]
[approve ci debian]

**Comment by dragon512** [comment]
My worry about changing the hash function from the "std" one is that this could have unexpected results with the std::string_view is used on systems that have it

**Comment by zwoop** [comment]
@dragon512 I suspect this PR depends on some other PR which is not on 7.1.x, can you confirm?

**Pull request merged**